### PR TITLE
chore: remove concat-stream

### PR DESCRIPTION
**Pull request by JustinBeckwith**
I'm just playing around with reducing our dependency graph.  I think I covered most of the useful stuff concat-stream was doing 🤷‍♂️ 

@stephenplusplus - do you know what kind of data these streams typically handle?  The tests just send through buffers, but it looks like concat-stream does a fair bit with type checking.  It would be super rad to know what exactly this does to make sure I ain't breaking it ⚒

### CODE REVIEW

File: src/paginator.ts [hunk 1]
```diff
@@ -184,13 +183,22 @@ export class Paginator {
...
       paginator.runAsStream_(parsedArguments, originalMethod)
         .on('error', callback)
-        .pipe(
-          concat((results: any) => {
-            callback(null, results);
-          })
-        );
+        .on('data', data => results.push(data))
+        .on('end', () => {
+          if (results.length > 0) {
```
Comment (stephenplusplus) [inline src/paginator.ts]: This section of code makes me want `concat-stream` back :) Whenever I see this pattern in user-land, I always want to ping the developer and say "Hey, y'know, there's this module called `concat-stream` that'll do that for ya."

I don't think we should flatten the dependencies too much, if it means re-inventing the wheel, thus defeating the utility of modularity that npm is good at.

**Comment by stephenplusplus** [comment]
The methods that use this are all `get{ServiceObject}s`, e.g. `storage.getBuckets()`. All of the things that are being concat'd are objects into an array, so really, this is all it's doing:

```js
const results = []

stream
  .on('data', result => results.push(result))
  .on('end', () => callback(null, results))
```

I suppose it could be considered overkill to use `concat-stream`, since it has more utility than we need. I can go either way :)

**Comment by JustinBeckwith** [comment]
> This section of code makes me want concat-stream back :) 
 😂I'll admit that I threw up in my mouth when I wrote it, but hey, it was a conversation starter  😂

> All of the things that are being concat'd are objects into an array
And *that* makes this actually worth doing.  I'll submit a patch 👍

File: test/paginator.ts [hunk 2]
```diff
@@ -349,27 +349,22 @@ describe('paginator', () => {
       });
 
       it('should return all results on end', (done) => {
-        const results = ['a', 'b', 'c'];
+        const results = [{a: 1}, {b: 2}, {c: 3}];
```
[the remark this answers is not in the document]
Comment (stephenplusplus) [reply test/paginator.ts]: Yep, this looks right to me!

**Pull request merged**